README.md
<img src='assets/nugget.png'>


# Open Source Clicking Game

This is an open source clicking game app being developed with Flutter. The app will have scoreboards, boosts, leaderboards, and other features common to clicking games.

## Status

The app is currently in the development stage and not yet released. Development is expected to take 2-3 months. Contributions are now accepted and feature requests are welcome.

<div align='center'>

  [![Status](https://img.shields.io/badge/Status-Under_Development-orange?style=for-the-badge)](https://shields.io/)
  [![Made with Flutter](https://img.shields.io/badge/Made_with-Flutter-blue?style=for-the-badge&logo=flutter)](https://flutter.dev/)
  ![Version](https://img.shields.io/badge/Version-0.2.6.8_(unreleased)-blue?style=for-the-badge&)
  ![FireBase](https://img.shields.io/badge/Backend-Firebase-red?style=for-the-badge&logo=firebase)
  ![Downloads](https://img.shields.io/badge/Beta_Downloads-3-red?style=for-the-badge&logo=Downloads)
  
  

</div>

## Features

- Scoreboards
- Boosts
- Leaderboards

## Feature Requests

If you have any ideas or suggestions for features you'd like to see in the app, please open an issue with the label "feature request". Your request will be reviewed and considered for inclusion in the app.

## License

This app is released under the terms of the GNU General Public License v3.0. You can find a copy of the license in the [LICENSE](LICENSE) file.

# How to Build the App


- Clone the repository to your local machine.
- Make sure you have the Flutter SDK installed and added to your PATH.
- Open the project in your preferred IDE (such as Android Studio or VS Code).
- Run ``flutter pub get`` to install the required dependencies.
- Run ``flutter run`` to launch the app on your connected device or emulator.

- You will also have to configure Firebase for any authentication.

NOTE: Steps to build the app may be different for you depending on the type of machine you are running on.





CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file in hierarchy of the most recent version and date.


## 0.2.4.4 [Unreleased] 7 June, 2023.

### Added
- New firebase email messages
- Firebase functions


### Changed
- None


### Fixed
- Email verification exploits




## 0.2.4.4 [Unreleased] 7 June, 2023.

### Added
- Complete working Login Page
- Complete working Sign Up Page (Minor Issues)
- Firebase integration
- New APK files for android that can be downloaded from /apks/apk-release.apk (20.9MB)


### Changed
- added more packages to pubspec.yaml
- Login page layout
- Added reference login page screenshots in /assets


### Fixed
- Few minor issues with Firebase Authentication




## 0.2.1 [Unreleased] 31 May, 2023.

### Added
- Login Page (No functionality).
- New sign in button on App Bar (may be changed)
- pubspec.yaml file

### Changed
- App Bar width due to new login button.
- imgs/ folder to assets/

### Fixed
- None





## 0.1.6 [Unreleased] 24 May, 2023.

### Added
- Settings page
- Icons turn white when on the respective page

### Changed
- App bar width

### Fixed
- None


## 0.1.4 [Unreleased] 21 May, 2023.

### Added
- Chicken Nugget in the middle of the screen
- Click Counter 
- Pop Effect upon click

### Changed
- Share Button to iOS share button (for now)
- Icons

### Fixed
- None


## 0.1.2 [Unreleased] 19 May, 2023.

### Added
- Main Home Screen

### Changed
- None

### Fixed
- None



## 0.1.1 [Unreleased] 27 April, 2023.

### Added
- CODE_OF_CONDUCT.md file 

### Changed
- README.md file

### Fixed
- None




## 0.1.0 [Unreleased]

### Added
- Initial project setup

### Changed
- None

### Fixed
- None


lib/sign_in_auth.dart
import 'package:flutter/material.dart';
import 'main.dart';

class SignIn extends StatelessWidget {
  const SignIn({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        backgroundColor: Colors.yellow
        ,
        appBar: AppBar(
          title: const Text('Sign in'),
          backgroundColor: Colors.brown,
        ),
        body: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [

              const Text("Let's Get You Started", style: TextStyle(fontSize: 35, fontWeight: FontWeight.bold,),),

              const SizedBox(height:150),

              Container(

                color: Colors.black,
                width:  360,
                child: TextButton(onPressed:  () {},
                    child: Row(
                      children: const [
                        Icon(Icons.apple, color: Colors.white,),
                        SizedBox(width: 80),
                        Text('Sign in with Apple', style: TextStyle(color: Colors.white),),

                      ],
                    ) ),
              ),
              const SizedBox(height: 10,),

              Container(

                color: Colors.white,
                width: 360,
                child: TextButton(onPressed:  () {},
                    child: Row(
                      children: const [
                        Icon(FontAwesomeIcons.google, color: Colors.black,),
                        SizedBox(width: 80),
                        Text('Sign in with Google', style: TextStyle(color: Colors.black),),
                      ],
                    ) ),
              ),

              SizedBox(height: 10,),

              Container(

                color: Colors.blueAccent,
                width: 360,
                child: TextButton(onPressed:  () {},
                    child: Row(
                      children: const [
                        Icon(FontAwesomeIcons.facebook, color: Colors.white,),
                        SizedBox(width: 80),
                        Text('Sign in with Facebook', style: TextStyle(color: Colors.white),),

                      ],
                    ) ),
              ),

              const SizedBox(height: 12),

              const Text('Psst! Login forms coming soon!', style: TextStyle(color: Colors.brown, fontWeight: FontWeight.bold, fontSize: 10),),


              const SizedBox(height: 300,),

              Container(

                color: Colors.white,
                width: 200,
                child: TextButton(onPressed:  () {},
                    child: Row(
                      children: const [
                        Icon(FontAwesomeIcons.github, color: Colors.black,),
                        SizedBox(width: 20),
                        Text('Support the project', style: TextStyle(color: Colors.black),),

                      ],
                    ) ),
              ),

              const SizedBox(height: 10,),

              Container(

                color: Colors.pink,
                width: 200,
                child: TextButton(onPressed:  () {},
                    child: Row(
                      children: const [
                        Icon(Icons.coffee, color: Colors.black,),
                        SizedBox(width: 20),
                        Text('Buy me a coffee', style: TextStyle(color: Colors.white),),

                      ],
                    ) ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

analytics/analytics.py
# Simple script used for analytics
# Requires manual run

import firebase_admin
from firebase_admin import credentials

cred = credentials.Certificate('keys/serviceAccountKey.json')
firebase_admin.initialize_app(cred)

from firebase_admin import analytics

# Retrieve user's data from firebase
def retrieve_user_data():
    users = analytics.get_all_users(start_date='2023-01-01', end_date='2023-06-01')
    for user in users:
        print('User ID:', user.user_id)
        print('First Access:', user.first_access)
        print('Last Access:', user.last_access)
        print('-----------------------------------')

 # Retrieve Event data
def retrieve_event_data():
    events = analytics.get_events(start_date='2023-01-01', end_date='2023-06-01')
    for event in events:
        print('Event Name:', event.name)
        print('Event Date:', event.date)
        print('Event User ID:', event.user_id)
        print('-----------------------------------')

retrieve_user_data()
retrieve_event_data()


email/functions/index.js
/**
 * Import function triggers from their respective submodules:
 *
 * const {onCall} = require("firebase-functions/v2/https");
 * const {onDocumentWritten} = require("firebase-functions/v2/firestore");
 *
 * See a full list of supported triggers at https://firebase.google.com/docs/functions
 */


// Create and deploy your first functions
// https://firebase.google.com/docs/functions/get-started

// exports.helloWorld = onRequest((request, response) => {
//   logger.info("Hello logs!", {structuredData: true});
//   response.send("Hello from Firebase!");
// });



// You will need the blaze plan for firebase in order to use this code. As well as firebase funtions


const functions = require('firebase-functions');
const admin = require('firebase-admin');
const fs = require('fs');
const path = require('templates/welcome_email.html');
admin.initializeApp();

exports.sendWelcomeEmail = functions.auth.user().onCreate((user) => {
  const userEmail = user.email;
  const fromEmail = functions.config().email.from;

  const delay = 60000; // 1 minute delay

  setTimeout(() => {
    const templatePath = path.join(__dirname, 'templates/welcome_email.html');
    const htmlTemplate = fs.readFileSync(templatePath, 'utf8');
    const customizedTemplate = htmlTemplate.replace('{{username}}', username);

    const mailOptions = {
      from: fromEmail,
      to: userEmail,
      subject: 'Welcome to The App',
      html: customizedTemplate,
    };

    admin
      .firestore()
      .collection('mail')
      .add(mailOptions)
      .then(() => {
        console.log('Welcome email sent successfully');
      })
      .catch((error) => {
        console.error('Error sending welcome email:', error);
      });
  }, delay);

  return null;
});

macos/Flutter/GeneratedPluginRegistrant.swift
//
//  Generated file. Do not edit.
//

import FlutterMacOS
import Foundation

import audioplayers
import cloud_firestore
import firebase_auth
import firebase_core
import path_provider_foundation
import shared_preferences_foundation
import sign_in_with_apple

func RegisterGeneratedPlugins(registry: FlutterPluginRegistry) {
  AudioplayersPlugin.register(with: registry.registrar(forPlugin: "AudioplayersPlugin"))
  FLTFirebaseFirestorePlugin.register(with: registry.registrar(forPlugin: "FLTFirebaseFirestorePlugin"))
  FLTFirebaseAuthPlugin.register(with: registry.registrar(forPlugin: "FLTFirebaseAuthPlugin"))
  FLTFirebaseCorePlugin.register(with: registry.registrar(forPlugin: "FLTFirebaseCorePlugin"))
  PathProviderPlugin.register(with: registry.registrar(forPlugin: "PathProviderPlugin"))
  SharedPreferencesPlugin.register(with: registry.registrar(forPlugin: "SharedPreferencesPlugin"))
  SignInWithApplePlugin.register(with: registry.registrar(forPlugin: "SignInWithApplePlugin"))
}


lib/main.dart

import 'package:flutter/material.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:firebase_auth/firebase_auth.dart';
import 'settings_page.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:font_awesome_flutter/font_awesome_flutter.dart';
import 'package:google_sign_in/google_sign_in.dart';
import 'package:shared_preferences/shared_preferences.dart';
import 'package:flutter/foundation.dart';
import 'package:flutter/services.dart';
import 'dart:async';

void main() async {

  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp();
  SharedPreferences prefs = await SharedPreferences.getInstance();
  bool isLoggedIn = prefs.getBool('isLoggedIn') ?? false;
  runApp(MaterialApp(home: isLoggedIn ? const ChickenNuggetClickerApp() : const SignIn(),
  ));
}

class ChickenNuggetClickerApp extends StatefulWidget {
  const ChickenNuggetClickerApp({Key? key}) : super(key: key);




@override
  _ChickenNuggetClickerAppState createState() => _ChickenNuggetClickerAppState();
}

class _ChickenNuggetClickerAppState extends State<ChickenNuggetClickerApp> {
  int counter = 0;
  bool isPopped = false;
  bool isSettingsPageActive = false;
  late User? _user;

  @override
  void initState() {
    super.initState();
    _initAuth();
  }

  Future<void> _initAuth() async {
    FirebaseAuth.instance.authStateChanges().listen((User? user) async {
      setState(() {
        _user = user;
      });

      if (_user != null) {
        // Load user's clicks from Firestore
        final docSnapshot = await FirebaseFirestore.instance
            .collection('users')
            .doc(_user!.uid)
            .collection('clicks')
            .get();

        setState(() {
          counter = docSnapshot.docs.length;
        });
      }
    });
  }


  void _handleButtonClick() {
    setState(() {
      counter++;
      isPopped = true;
    });

    Future.delayed(const Duration(milliseconds: 30), () {
      setState(() {
        isPopped = false;
      });
    });

    if (_user != null) {
      // Log click to Firestore
      FirebaseFirestore.instance.collection('users').doc(_user!.uid).collection('clicks').add({
        'timestamp': DateTime.now(),
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          backgroundColor: Colors.brown,
          title: const Text('Click!'),
          leadingWidth: 90,
          toolbarHeight: 60,
          leading: Container(
            margin: const EdgeInsets.only(left: 2),
            child: IconButton(
              icon: const Icon(Icons.ios_share_outlined, color: Colors.black, size: 40,),
              onPressed: () {},
            ),
          ),
          actions: [
            IconButton(
              icon: const Icon(Icons.login_outlined, color: Colors.black, size: 30,),
              onPressed: () {
                Navigator.push(
                  context,
                  MaterialPageRoute(builder: (_) => const SignIn()),
                );

              }
            ),
          ],
        ),
        backgroundColor: const Color.fromRGBO(199, 144, 83, 1),
        body: Builder(
          builder: (BuildContext context) {
            return Column(
              children: [
                Align(
                  child: Padding(
                    padding: const EdgeInsets.only(top: 160.0),
                    child: Column(
                      mainAxisAlignment: MainAxisAlignment.start,
                      crossAxisAlignment: CrossAxisAlignment.center,
                      children: [
                        GestureDetector(
                          onTap: _handleButtonClick,
                          child: AnimatedContainer(
                            duration: const Duration(milliseconds: 30),
                            width: isPopped ? 280 : 300,
                            height: isPopped ? 280 : 300,
                            child: Padding(
                              padding: const EdgeInsets.only(bottom: 40),
                              child: Image.asset(
                                'assets/nugget.png',
                                fit: BoxFit.cover,
                              ),
                            ),
                          ),
                        ),
                        const SizedBox(height: 2),
                        Text(
                          isPopped ? 'Nuggets: $counter' : 'Nuggets: $counter',
                          style: const TextStyle(
                            fontSize: 20,
                            fontWeight: FontWeight.bold,
                            color: Colors.white,
                          ),
                        ),
                      ],
                    ),
                  ),
                ),
                Expanded(
                  child: Align(
                    alignment: Alignment.bottomCenter,
                    child: Container(
                      color: Colors.brown,
                      padding: const EdgeInsets.all(10),
                      child: Row(
                        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                        children: [
                          Expanded(
                            child: Container(
                              decoration: BoxDecoration(
                                borderRadius: BorderRadius.circular(50),
                              ),
                              height: 60,
                              child: TextButton(
                                style: ButtonStyle(
                                  shape: MaterialStateProperty.all<CircleBorder>(
                                    const CircleBorder(),
                                  ),
                                ),
                                onPressed: () {},
                                child: const Icon(
                                  Icons.ads_click_outlined,
                                  size: 40,
                                  color: Colors.black,
                                ),
                              ),
                            ),
                          ),
                          Expanded(
                            child: Container(
                              color: Colors.brown,
                              height: 60,
                              child: TextButton(
                                onPressed: () {},
                                style: ButtonStyle(
                                  shape: MaterialStateProperty.all<CircleBorder>(
                                    const CircleBorder(),
                                  ),
                                ),
                                child: const Icon(
                                  Icons.leaderboard_outlined,
                                  size: 40,
                                  color: Colors.black,
                                ),
                              ),
                            ),
                          ),
                          Expanded(
                            child: Container(
                              color: Colors.brown,
                              height: 60,
                              child: TextButton(
                                onPressed: () {
                                  setState(() {
                                    isSettingsPageActive = true;
                                  });
                                  Navigator.pushReplacement(
                                    context,
                                    MaterialPageRoute(
                                      builder: (_) => const SettingsPage(),
                                    ),
                                  );
                                },
                                style: ButtonStyle(
                                  shape: MaterialStateProperty.all<CircleBorder>(
                                    const CircleBorder(),
                                  ),
                                ),
                                child: Icon(
                                  Icons.settings_outlined,
                                  size: 40,
                                  color: isSettingsPageActive ? Colors.white : Colors.black,
                                ),
                              ),
                            ),
                          ),
                        ],
                      ),
                    ),
                  ),
                ),
              ],
            );
          },
        ),
      ),
    );
  }
}

class SignIn extends StatefulWidget {
  const SignIn({Key? key}) : super(key: key);

  @override
  _SignInState createState() => _SignInState();
}

class _SignInState extends State<SignIn> {
  final _formKey = GlobalKey<FormState>();
  final FirebaseAuth _auth = FirebaseAuth.instance;
  late String _email;
  late String _password;


  @override
  void initState() {
    super.initState();
    _email = '';
    _password = '';
  }

  void _signInWithEmailAndPassword() async {
    if (_formKey.currentState!.validate()) {
      _formKey.currentState!.save();

      try {
        await _auth.signInWithEmailAndPassword(
          email: _email,
          password: _password,
        );

        SharedPreferences prefs = await SharedPreferences.getInstance();
        await prefs.setBool('isLoggedIn', true);

        Navigator.pushReplacement(
          context,
          MaterialPageRoute(builder: (_) => const ChickenNuggetClickerApp()),
        );
      } catch (e) {
        showDialog(
          context: context,
          builder: (BuildContext context) {
            return AlertDialog(
              title: const Text('Sign-in Error'),
              content: Text(e.toString()),
              actions: [
                TextButton(
                  onPressed: () => Navigator.of(context).pop(),
                  child: const Text('OK'),
                ),
              ],
            );
          },
        );
      }
    }
  }

  void _goToSignUp() {
    Navigator.pushReplacement(
      context,
      MaterialPageRoute(builder: (_) => const SignUp()),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.yellow,
      appBar: AppBar(
        backgroundColor: Colors.brown,
        title: const Text('Sign In'),
      ),
      body: SingleChildScrollView(
        child: Padding(
          padding: const EdgeInsets.all(16.0),
          child: Form(
            key: _formKey,
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                Image.asset('assets/nugget.png'),
                const SizedBox(height: 20),
                const Text(
                  "Welcome back!",
                  style: TextStyle(fontSize: 30),
                ),
                const SizedBox(height: 30),
                TextFormField(
                  keyboardType: TextInputType.emailAddress,
                  onChanged: (value) => _email = value.trim(),
                  validator: (value) {
                    if (value == null || value.isEmpty) {
                      return 'Please enter your email';
                    }
                    return null;
                  },
                  style: const TextStyle(color: Colors.black),
                  decoration: InputDecoration(
                    hintText: 'Enter your email',
                    filled: true,
                    fillColor: Colors.white,
                    border: OutlineInputBorder(
                      borderSide: BorderSide.none,
                      borderRadius: BorderRadius.circular(8.0),
                    ),
                  ),
                ),
                const SizedBox(height: 16.0),
                TextFormField(
                  obscureText: true,
                  onChanged: (value) => _password = value,
                  validator: (value) {
                    if (value == null || value.isEmpty) {
                      return 'Please enter your password';
                    }
                    return null;
                  },
                  style: const TextStyle(color: Colors.black),
                  decoration: InputDecoration(
                    hintText: 'Enter your password',
                    filled: true,
                    fillColor: Colors.white,
                    border: OutlineInputBorder(
                      borderSide: BorderSide.none,
                      borderRadius: BorderRadius.circular(8.0),
                    ),
                  ),
                ),
                const SizedBox(height: 16.0),
                ElevatedButton(
                  onPressed: _signInWithEmailAndPassword,
                  child: const Row(
                    children: [
                      SizedBox(width: 160),
                      Text('Sign in'),
                    ],
                  ),
                ),
                const SizedBox(height: 80),
                const Text(
                  'Don\'t have an account?',
                  style: TextStyle(fontSize: 16),
                ),
                const SizedBox(height: 14),
                Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    TextButton(
                      onPressed: _goToSignUp,
                      child: const Text('Sign up'),
                    ),
                  ],
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}





// SIGN UP PAGE BELOW




class SignUp extends StatefulWidget {
  const SignUp({Key? key}) : super(key: key);

  @override
  _SignUpState createState() => _SignUpState();
}

class _SignUpState extends State<SignUp> {
  final _formKey = GlobalKey<FormState>();
  late String _email;
  late String _password;
  final FirebaseAuth _auth = FirebaseAuth.instance;
  bool isEmailVerified = false;
  late Timer _timer;

  @override
  void initState() {
    super.initState();
    // Start the timer when the widget is initialized
    _timer = Timer.periodic(const Duration(seconds: 3), (timer) {
      checkEmailVerificationStatus();
    });
  }

  @override
  void dispose() {
    // Cancel the timer when the widget is disposed
    _timer.cancel();
    super.dispose();
  }

  void checkEmailVerificationStatus() async {
    final User? currentUser = _auth.currentUser;
    await currentUser?.reload();
    if (currentUser != null && currentUser.emailVerified) {
      setState(() {
        isEmailVerified = true;
      });
      _timer.cancel();
      // Navigate to success page or perform any other actions
      Navigator.pushReplacement(
        context,
        MaterialPageRoute(builder: (_) => const SuccessPage()),
      );
    }
  }

  Future<void> _signUpWithEmailAndPassword() async {
    if (_formKey.currentState!.validate()) {
      _formKey.currentState!.save();

      try {
        final UserCredential userCredential =
        await _auth.createUserWithEmailAndPassword(
          email: _email,
          password: _password,
        );

        await userCredential.user!.sendEmailVerification();

        showDialog(
          context: context,
          barrierDismissible: false,
          builder: (BuildContext context) {
            return const AlertDialog(
              title: Text('Email Verification'),
              content: Text(
                'A verification email has been sent to your email address. '
                    'Please click the verification link to proceed.',
              ),
            );
          },
        );
      } catch (e) {
        showDialog(
          context: context,
          builder: (BuildContext context) {
            return AlertDialog(
              title: const Text('Sign-up Error'),
              content: Text(e.toString()),
              actions: [
                TextButton(
                  onPressed: () => Navigator.of(context).pop(),
                  child: const Text('OK'),
                ),
              ],
            );
          },
        );
      }
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.yellow,
      appBar: AppBar(
        backgroundColor: Colors.brown,
        title: const Text('Sign Up'),
      ),
      body: SingleChildScrollView(
        child: Padding(
          padding: const EdgeInsets.all(16.0),
          child: Form(
            key: _formKey,
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                Image.asset('assets/nugget.png'),
                const SizedBox(height: 20),
                const Text(
                  "Welcome!",
                  style: TextStyle(fontSize: 30),
                ),
                const SizedBox(height: 30),
                TextFormField(
                  keyboardType: TextInputType.emailAddress,
                  onChanged: (value) => _email = value.trim(),
                  validator: (value) {
                    if (value == null || value.isEmpty) {
                      return 'Please enter your email';
                    }
                    return null;
                  },
                  style: const TextStyle(color: Colors.black),
                  decoration: InputDecoration(
                    hintText: 'Enter your email',
                    filled: true,
                    fillColor: Colors.white,
                    border: OutlineInputBorder(
                      borderSide: BorderSide.none,
                      borderRadius: BorderRadius.circular(8.0),
                    ),
                  ),
                ),
                const SizedBox(height: 16.0),
                TextFormField(
                  obscureText: true,
                  onChanged: (value) => _password = value,
                  validator: (value) {
                    if (value == null || value.isEmpty) {
                      return 'Please enter your password';
                    }
                    return null;
                  },
                  style: const TextStyle(color: Colors.black),
                  decoration: InputDecoration(
                    hintText: 'Enter your password',
                    filled: true,
                    fillColor: Colors.white,
                    border: OutlineInputBorder(
                      borderSide: BorderSide.none,
                      borderRadius: BorderRadius.circular(8.0),
                    ),
                  ),
                ),
                const SizedBox(height: 16.0),
                ElevatedButton(
                  onPressed: _signUpWithEmailAndPassword,
                  child: const Row(
                    children: [
                      SizedBox(width: 160),
                      Text('Sign up'),
                    ],
                  ),
                ),
                const SizedBox(height: 80),
                const Text(
                  'Or Continue with...',
                  style: TextStyle(fontSize: 16),
                ),
                const SizedBox(height: 14),
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                  children: [
                    TextButton(
                      style: ButtonStyle(
                        backgroundColor: MaterialStateProperty.all(Colors.black),
                      ),
                      child: const Row(
                        children: [
                          SizedBox(
                            height: 70,
                            width: 20,
                          ),
                          Icon(
                            Icons.apple,
                            color: Colors.white,
                            size: 40,
                          ),
                          SizedBox(
                            width: 20,
                          ),
                        ],
                      ),
                      onPressed: () {},
                    ),
                    TextButton(
                      style: ButtonStyle(
                        backgroundColor: MaterialStateProperty.all(Colors.white),
                      ),
                      child: const Row(
                        children: [
                          SizedBox(
                            height: 70,
                            width: 20,
                          ),
                          Icon(
                            FontAwesomeIcons.google,
                            color: Colors.black,
                            size: 40,
                          ),
                          SizedBox(width: 20)
                        ],
                      ),
                      onPressed: () {},
                    ),
                    TextButton(
                      style: ButtonStyle(
                        backgroundColor: MaterialStateProperty.all(Colors.blue),
                      ),
                      child: const Row(
                        children: [
                          SizedBox(
                            height: 70,
                            width: 20,
                          ),
                          Icon(
                            Icons.facebook,
                            color: Colors.white,
                            size: 40,
                          ),
                          SizedBox(
                            width: 20,
                          ),
                        ],
                      ),
                      onPressed: () {},
                    ),
                  ],
                ),
                Row(
                  children: [
                    const SizedBox(width: 100),
                    const Text("Already have an account?"),
                    TextButton(
                      onPressed: () {
                        Navigator.pushReplacement(
                          context,
                          MaterialPageRoute(
                            builder: (_) => const SignIn(),
                          ),
                        );
                      },
                      child: const Text('Sign in'),
                    ),
                  ],
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}




class SuccessPage extends StatelessWidget {
  const SuccessPage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final User? currentUser = FirebaseAuth.instance.currentUser;

    return Scaffold(backgroundColor: Colors.yellow,
      appBar: AppBar(
        backgroundColor: Colors.brown,
        title: const Text('Success!'),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            currentUser!.emailVerified
                ? const Text(
              'Sign up successful!',
              style: TextStyle(fontSize: 24),
            )
                : const Text(
              'Sign up successful!\n'
                  'Please verify your email to proceed.',
              textAlign: TextAlign.center,
              style: TextStyle(fontSize: 24),
            ),
            const SizedBox(height: 20),
            ElevatedButton(
              onPressed: () {
                Navigator.pushReplacement(
                  context,
                  MaterialPageRoute(
                    builder: (_) => const SignIn(),
                  ),
                );
              },
              child: const Text('Go to Sign In'),
            ),
          ],
        ),
      ),
    );
  }
}
